usaspending_api/accounts/helpers.py

```diff
@@ -16,3 +31,33 @@ def start_and_end_dates_from_fyq(fiscal_year, fiscal_quarter):
         end_date = datetime.date(fiscal_year, 9, 30)
 
     return start_date, end_date
+
+
+def build_tas_codes_filter(queryset, model, value):
+    """
+    Build the TAS codes filter.  Because of performance issues, the normal
+    trick of checking for award_id in a subquery wasn't cutting it.  To
+    work around this, we're going to use the query.extra function to add SQL
+    that should give us a better query plan.
+    """
+
+    # Build the filtering for the tas_award_matview.
+    or_queryset = Q()
+    for tas in value:
+        or_queryset |= Q(**{TAS_COMPONENT_TO_FIELD_MAPPING[k]: v for k, v in tas.items()})
+
+    if or_queryset:
+        # Now that we've built the actual filter, let's turn it into SQL that we
+        # can provide to the queryset.extra method.  We do this by converting the
+        # queryset to raw SQL.
+        where_sql, where_params = generate_where_clause(TASAwardMatview.objects.filter(or_queryset))
+        return queryset.extra(
+            tables=[TASAwardMatview._meta.db_table],
+            where=[
+                '"{}".award_id = "{}".award_id'.format(TASAwardMatview._meta.db_table, model._meta.db_table),
+                where_sql,
+            ],
+            params=where_params
+        )
+
+    return queryset
```

Comment:
Why is this in the accounts folder and not awards or search?

Reply:
presumably because TAS = Treasury **Account** Symbol

Reply:
I don't know.  It's very confusing.  There's stuff scattered everywhere for this set of endpoints/models.  I just put it with the constants.  Once again, I don't know how much I care.  If you have a preference, let it be known... awards or search or somewhere else entirely.

Reply:
I think it should be in search for cleaner encapsulation. I can do it myself if that would be easier for you (and won't make a bunch of code conflicts).